### PR TITLE
test: Add e2e tests for sink update

### PR DESCRIPTION
**Pull request by jachym-tousek-keboola**
Jira: https://keboola.atlassian.net/browse/PSGO-670 (second part)

**Changes:**
- Add e2e test for sink update

-----------

### CODE REVIEW

File: test/stream/api/sink/update-005-missing-storage-token-400/001-update-source/expected-response.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "statusCode": 400,
```
Comment (Matovidlo) [inline test/stream/api/sink/update-005-missing-storage-token-400/001-update-source/expected-response.json:2]: Isn't this `401` error?

Comment (michaljurecko) [reply test/stream/api/sink/update-005-missing-storage-token-400/001-update-source/expected-response.json]: The GOA framework takes it as a invalid/missing input, and that's why it's `400 - bad request`. 

Some middleware would be required to change the code.

Comment (michaljurecko) [reply test/stream/api/sink/update-005-missing-storage-token-400/001-update-source/expected-response.json]: image

**Review by michaljurecko — COMMENTED** [review]
I don't think I need to check it, tag me if you have any questions

**Review by hosekpeter — APPROVED** [review]

**Review by Matovidlo — APPROVED** [review]
LGTM

File: test/stream/api/sink/update-005-missing-storage-token-400/001-update-source/expected-response.json (hunk 1 above)
Comment (michaljurecko) [reply test/stream/api/sink/update-005-missing-storage-token-400/001-update-source/expected-response.json]: GOA inputs are:
- headers
- url params
- post body fields

`APIKey` is required payload field provided by the `X-StorageApi-Token` header.

https://github.com/keboola/keboola-as-code/blob/0446f5ef527fff1837b5891584789f5e48d65a77/api/stream/design.go#L117-L119

https://github.com/keboola/keboola-as-code/blob/edbae4bf38a014ee30d17b98d1f1d429dc24d679/internal/pkg/service/common/goaextension/token/token.go#L101-L118

Comment (michaljurecko) [reply test/stream/api/sink/update-005-missing-storage-token-400/001-update-source/expected-response.json]: You can remove the `Required` definition there:
https://github.com/keboola/keboola-as-code/blob/edbae4bf38a014ee30d17b98d1f1d429dc24d679/internal/pkg/service/common/goaextension/token/token.go#L104

And return better `401` error there:
https://github.com/keboola/keboola-as-code/blob/edbae4bf38a014ee30d17b98d1f1d429dc24d679/internal/pkg/service/stream/api/service/auth.go#L15-L20

Please, store somewhere link to this notes.

**Pull request merged**